### PR TITLE
Issues/3285 update reader analytics

### LINKED ISSUE #3285: Add missing reader analytics
Reader analytics in the Android app are out of sync with the iOS app now that [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/4308) has been merged.

### PR DESCRIPTION
**Pull request by aerych**
Closes #3285 

Adds new analytics to the reader and tweaks some of the naming to better match the tracks convention.
Some of the added keys and calls related to discover and lists are in anticipation of future feature support. I've included them now to match what's in the iOS app.  

Side note: I noticed there are a couple of inconsistencies in the iOS app (calls recently added to the iOS app that previously were made by the Android app) that need to be addressed in *that* repo and I have a branch waiting to merge once the iOS analytics pod is updated.  After that's done the two should be synced up.

Needs review: @nbradbury

### CODE REVIEW

**Comment by nbradbury** [comment]
@aerych Did you mean for this to be for 4.7 instead of 4.6? I ask because we're beyond code freeze for 4.6 (check the P2 sidebar), although we can still submit 4.6 bug fixes. If it really is intended for 4.6, then it'll need to be branched from `release/4.6` rather than `develop`.

File: WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java [hunk 1]
```diff
@@ -157,6 +157,14 @@ public boolean isFollowedSites() {
         return tagType == ReaderTagType.DEFAULT && getEndpoint().endsWith("/read/following");
     }
     public boolean isFreshlyPressed() {
-        return tagType == ReaderTagType.DEFAULT && getTagName().endsWith("/freshly-pressed");
+        return tagType == ReaderTagType.DEFAULT && getEndpoint().endsWith("/freshly-pressed");
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderTag.java]: :scream: 

Good catch - thanks for fixing my bug!

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 2]
```diff
@@ -1006,16 +1006,30 @@ public void onTagSelected(String tagName) {
...
-        AnalyticsTracker.track(AnalyticsTracker.Stat.READER_LOADED_TAG, properties);
-        if (tag.isFreshlyPressed()) {
-            AnalyticsTracker.track(AnalyticsTracker.Stat.READER_LOADED_FRESHLY_PRESSED);
-        }
+        trackTagLoaded(tag);
         AppLog.d(T.READER, String.format("reader post list > tag %s displayed", tag.getTagNameForLog()));
         setCurrentTag(tag);
     }
 
+    public void trackTagLoaded(ReaderTag tag) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java]: Minor nit: this method can be `private` rather than `public`.

**Comment by aerych** [comment]
> Did you mean for this to be for 4.7
Oopsie! Updating.

**Comment by nbradbury** [comment]
Looks good :shipit:

**Pull request merged**